pkg/controller/hawtio/certificate.go

```
@@ -83,3 +86,66 @@ func generateCertificateSecret(name string, namespace string, caSecret *corev1.S
 		}, Type: corev1.SecretTypeTLS,
 	}, nil
 }
+
+func ValidateCertificate(caSecret corev1.Secret, validAtLeastForHours float64) bool {
+	block, _ := pem.Decode(caSecret.Data[corev1.TLSCertKey])
+	cert, err := x509.ParseCertificate(block.Bytes)
+
+	if err != nil {
+		log.Error(err, "certificate reading error")
```

I'd suggest to return the error. Is there a valid reason why the certificate could not be parsed?

---

I believe only if user will manually mess up something.

---

Right, I think that's better to return the error so that it's logged in that case and not ignored.